docs/global/modules/core/pages/howto/modelql-writing.adoc

```diff
@@ -0,0 +1,53 @@
+= Writing via ModelQL
+
+== Writing with the type-safe ModelQL API
+
+The extensions generated by `model-api-gen-gradle` also include functions for write access.
+These can be called on receivers that are instance of `IMonoStep`.
+
+[NOTE]
+--
+Specify the link:../reference/component-model-api-gen-gradle.adoc#model-api-gen-gradle_attributes_modelqlKotlinDir[modelqlKotlinDir] property to enable the generation.
+--
+
+For properties, references and single-valued child links use the `set`-functions.
+
+[source,kotlin]
+--
+client.query { root ->
+    root.children("classes").ofConcept(C_ClassConcept)
+        .member
+        .ofConcept(C_StaticMethodDeclaration)
+        .first()
+        .setName("myNewMethodName".asMono())
+}
+--
+
+To add a new child to multi-valued child links use the generated `addTo`-functions.
+The index parameter is optional: by default the child will be appended at the end.
+In case the target type of the child link is not abstract, the concept parameter can be omitted, because a default value will have been generated.
+
+[source,kotlin]
+--
+client.query { root ->
+    root.children("classes").ofConcept(C_ClassConcept)
+        .first()
+        .addToMember(C_StaticMethodDeclaration, 1)
+}
+--
+
+To remove a node, you will need to use the untyped API by calling `.untyped().remove()` on the typed node you want to remove.
```

Comment:
Is there any good reason for not passing through the `remove` method?

Reply:
I don't think there is. It shouldn't be part of the generated API but it should be possible for us to just add an extension function for `IMonoStep<TypedNode>` called `remove` which calls `untyped().remove()`.